### ISSUE TITLE
Document use with LuaJIT

### ISSUE THREAD

**Issue by rrthomas** [keplerproject/lua-compat-5.2#19]
It may be that I failed to read between the lines, but it's not obvious to me what this package does for LuaJIT. For example, the recently-added support for `luaL_Stream`. Is it intended that all the supported C APIs are available on LuaJIT? Only some of them?

**Comment by siffiejoe** [comment]
`Compat-5.2` was mainly developed and tested for PUC-Rio Lua 5.1. That said, LuaJIT is supposed to be binary compatible with Lua 5.1, and most reimplemented C APIs use only the official Lua 5.1 C API, so all functions should work fine. Problem areas are the above mentioned `luaL_Stream` support (where LuaJIT does things differently than PUC-Rio Lua 5.1), `luaL_traceback` and `luaL_fileresult` (which LuaJIT provides on its own), and some semi-official macros from `luaconf.h` which LuaJIT doesn't have (but we provide fallbacks). I have compiled, linked, and tested `compat-5.2.c` using LuaJIT. However, the Lua part is a different story: LuaJIT already provides a lot of Lua 5.2 compatibility (especially if compiled with `LUAJIT_ENABLE_LUA52COMPAT`), and some of it interferes with our implementations.

TL;DR: If `compat-5.2.c` doesn't work for LuaJIT,  please file a bug report and I will try to fix it.

**Comment by rrthomas** [comment]
It would be great to have that TL;DR in the README!

**Comment by siffiejoe** [comment]
I've added a small note to the README.
